sound/soc/sof/intel/hda.c

```diff
@@ -1176,8 +1177,15 @@ int hda_dsp_remove(struct snd_sof_dev *sdev)
 				SOF_HDA_INT_CTRL_EN | SOF_HDA_INT_GLOBAL_EN, 0);
 
 	/* disable cores */
-	if (chip)
-		hda_dsp_core_reset_power_down(sdev, chip->host_managed_cores_mask);
+	if (chip && chip->power_down_dsp) {
+		ret = chip->power_down_dsp(sdev);
+		if (ret) {
+			dev_err(sdev->dev, "error: power_down_dsp op returns %d\n", ret);
+			/* Even with the error, no return. dsp need to be disabled.
+			 * Need to go through to the end of the function.
```

Comment:
/* keep going without returning the error code, the DSP needs to be disabled */

Reply:
slightly re-phased, can you check?